File: CS-Examples/03_Cells/SetDBNumFormatting.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace SetDBNumFormatting
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Create an empty worksheet
            workbook.CreateEmptySheets(1);

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set values for cells A1, A2, and A3
            sheet.Range["A1"].Value2 = 123;
            sheet.Range["A2"].Value2 = 456;
            sheet.Range["A3"].Value2 = 789;

            // Get the cell range A1:A3
            CellRange range = sheet.Range["A1:A3"];

            // Set the DB num format for the range
            range.NumberFormat = "[DBNum2][$-804]General";

            // Auto fit columns for the range
            range.AutoFitColumns();

            // Save the modified workbook to a file
            string output = "SetDBNumFormatting_out.xlsx";
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the Excel file
            ExcelDocViewer(output);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: CS-Examples/03_Cells/SetDBNumFormatting.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace SetDBNumFormatting
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Creates a new instance of the Workbook class.
            Dim workbook As New Workbook()
            'Creates one empty worksheet in the workbook.
            workbook.CreateEmptySheets(1)

            'Retrieves the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Sets the value of cell A1 to 123.
            sheet.Range("A1").Value2 = 123
            'Sets the value of cell A2 to 456.
            sheet.Range("A2").Value2 = 456
            'Sets the value of cell A3 to 789.
            sheet.Range("A3").Value2 = 789

            'Specifies the range of cells from A1 to A3.
            Dim range As CellRange = sheet.Range("A1:A3")

            'Sets the number format of the range to a specified DBNum format.
            range.NumberFormat = "[DBNum2][$-804]General"

            'Adjusts the width of the columns in the range to fit the content.
            range.AutoFitColumns()

            'Specifies the name of the output file.
            Dim output As String = "SetDBNumFormatting_out.xlsx"
            'Saves the workbook to the specified file in Excel 2013 format.
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the Excel file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
